homeassistant/components/enocean/light.py

```diff
@@ -56,30 +71,60 @@ class EnOceanLight(EnOceanEntity, LightEntity):
     _attr_brightness = 50
     _attr_is_on = False
 
-    def __init__(self, sender_id: list[int], dev_id: list[int], dev_name: str) -> None:
+    def __init__(
+        self,
+        sender_id: list[int],
+        dev_id: list[int],
+        dev_name: str,
+        command_type: CentralCommandType,
+    ) -> None:
         """Initialize the EnOcean light source."""
         super().__init__(dev_id)
         self._sender_id = sender_id
         self._attr_unique_id = str(combine_hex(dev_id))
         self._attr_name = dev_name
+        self._command_type = command_type
+        if command_type == CentralCommandType.SWITCHING:
+            self._attr_color_mode = ColorMode.ONOFF
+            self._attr_supported_color_modes = {ColorMode.ONOFF}
 
-    def turn_on(self, **kwargs: Any) -> None:
-        """Turn the light source on or sets a specific dimmer value."""
+    def __dimming_command_on(self, kwargs: Any) -> list[int]:
         if (brightness := kwargs.get(ATTR_BRIGHTNESS)) is not None:
             self._attr_brightness = brightness
 
         bval = math.floor(self._attr_brightness / 256.0 * 100.0)
         if bval == 0:
-            bval = 1
-        command = [0xA5, 0x02, bval, 0x01, 0x09]
+            bval = 128
+        return [0xA5, 0x02, bval, 0x01, 0x09]
+
+    def __switching_command_on(self, kwargs: Any) -> list[int]:
+        return [0xA5, 0x01, 0x00, 0x00, 0x09]
```

Comment:
This is too much protocol details. Those should be kept in the 3rd party library.